full_history_avd/src/history_tree/mod.rs

```diff
@@ -418,11 +428,11 @@ mod tests {
     impl MerkleTreeAVDParameters for MerkleTreeAVDTestParameters {
         const MAX_UPDATE_BATCH_SIZE: u64 = 4;
         const MAX_OPEN_ADDRESSING_PROBES: u8 = 2;
-        type MerkleTreeParameters = MerkleTreeTestParameters;
+        type Storer = MemStore<MerkleTreeTestParameters>;
     }
 
     type TestMerkleTreeAVD = MerkleTreeAVD<MerkleTreeAVDTestParameters>;
-    type TestAVDWithHistory = SingleStepAVDWithHistory<TestMerkleTreeAVD, MerkleTreeTestParameters>;
+    type TestAVDWithHistory = SingleStepAVDWithHistory<TestMerkleTreeAVD, dyn Storer<P = MerkleTreeTestParameters>>;
 
     #[test]
     fn lookup_test() {
```

Comment:
@nirvantyagi 

`let (ssavd_pp, crh_pp) = TestAVDWithHistory::setup(&mut rng).unwrap();` gives compiler error. I'm pretty lost in the abstraction layers here. If we're making a `TestAVDWithHistory` I would expect some process like:

1) Make some type of Storer with some type of Hasher
2) Make a History tree with that Storer
3) Make the AVDWithHistory using that History tree

but I'm not understanding if that's what's happening

Reply:
This might be resolved once the above comment on fixing the concrete type of ``TestAVDWithHistory`` is addressed. Yeah totally understand, the abstractions are...a lot :see_no_evil: